src/Services/PatientService.php

```diff
@@ -963,4 +968,60 @@ private function getPatientSuffixKeys()
         }
         return $this->patientSuffixKeys;
     }
+
+    public function touchRecentPatientList(array $patient)
+    {
+        $user = new UserService();
+        $curUser = $user->getCurrentlyLoggedInUser();
+
+        $query = "SELECT patients FROM recent_patients WHERE user_id = ?";
+        $res = sqlStatement($query, $curUser['id']);
+        $row = sqlFetchArray($res);
+        $rp = ($row) ? unserialize($row['patients']) : [];
+
+        // In case we are returning to an already recently viewed patient, drop them from the current position
+        if (array_key_exists((string) $patient['pid'], $rp)) {
+            unset($rp[(string) $patient['pid']]);
+        }
+
+        // Get the columns we are storing
+        $query = "SELECT * FROM list_options WHERE list_id = 'recent_patient_columns' and activity = '1'";
+        $res = sqlStatement($query);
+        $cols = ['pid'];
+        while ($row = sqlFetchArray($res)) {
+            $cols[] = $row['option_id'];
+        }
+
+        // Trim down the incoming patient array to just the whitelisted columns
+        foreach ($patient as $k => $v) {
+            if (!in_array($k, $cols)) {
+                unset($patient[$k]);
+            }
+        }
+
+        // Push the new patient to the front of the FIFO list
+        array_unshift($rp, $patient);
+
+        // Cap out at max count as set in globals
+        if (count($rp) == ($GLOBALS['recent_patient_count'] + 1)) {
```

Comment:
would also do a >= in case it somehow is above (which would require popping the amount over). Could make this happen if go from a larger global to a smaller global and if that is the case then supporting >= would deal with that.